File: 1001.c
#include<stdio.h>

int main()
{
  int input=0, cnt=0;
  scanf("%d", &input);
    while(input != 1){
        if(input % 2){
            input=(3 * input + 1) / 2;
        }else{
            input = input / 2;
        }
        cnt++;
    }
  printf("%d", cnt);
  return 0;
 } 

File: 1001.go
package main

import "fmt"

func main(){
  cnt := 0
  var n int
  fmt.Scanln(&n)
  
  for n != 1{
    if n % 2 == 0{
      n /= 2
    } else{
      n = (3 * n + 1) / 2
    }
    cnt += 1
  }
  fmt.Println(cnt)
}